merkle/tree_hasher.go

```diff
@@ -14,11 +14,24 @@
 
 package merkle
 
-// TreeHasher is the interface that the previous tree hasher struct implemented.
+// TreeHasher computes generic merkle trees.
```

Comment:
This doc comment doesn't really make sense. The previous one didn't either. Should talk about providing hash computations for an implementation of Merkle tree hashing or something like that.